### PR TITLE
java.time generators

### PR DESCRIPTION
**Pull request by ivern**
Here's a partial implementation of the java.time.* generators, addressing issue #71.  At the moment, only Clock, Instant, Duration, LocalDate, LocalTime, and LocalDateTime are supported.  I figured I'd send you a PR now so you can have a look at how things are going.   I should have this wrapped up in another day or two.

Let me know if you have any comments or concerns.  Thanks.

### CODE REVIEW

**Comment by ivern** [comment]
Actually I'm going to go ahead and close this now--there's a very dumb bug in how I'm generating second/nanosecond tuples.  No point in reviewing until that's fixed, IMO.

**Comment by ivern** [comment]
PR reopened for review, sorry about that.  I've changed my approach to generating two-part time objects.  Now the code constructs single BigIntegers and delegates generation to the BigInteger generator.  I'd definitely appreciate any thoughts on whether that's a good idea, but it seems clean enough.

**Comment by pholser** [comment]
So far so good!

Looking at each of the `configure(InRange)` methods, I'm wondering if we can find a way to eliminate the duplication across each of them.

Also: four spaces is the unit of indentation; I generally don't declare locals `final` unless necessary.

**Comment by ivern** [comment]
I'll do cleanup before I'm done and handle the indentation issue + finals.  As for `configure(InRange)`, I'd considered refactoring these methods to eliminate duplication but this runs into a few issues:

1. The parse methods are static (this isn't too bad, I can set up the generators with a method reference to the parse method to call.)
2. We have three cases: configure with custom formatting, configure ignoring custom formatting, and configure with custom range comparisons (for `Period`, which isn't `Comparable`).   Still not too terrible.
3. Each of these configure methods behaves slightly differently, in particular regarding how the min/max strings are parsed.  I think there's value in documenting these differences.

Basically, the conclusion I reached is that we don't gain a whole lot by trying to refactor the configure methods.  However, I'll keep an eye on them and clean them up before I wrap up these generators if I can come up with a good way of doing this.

**Comment by ivern** [comment]
By the way, have you considered adding an .editorconfig file to the project?  It's supported by all major IDEs/editors, and is pretty awesome for keeping things like indentation consistent: http://editorconfig.org/

**Comment by pholser** [comment]
Good to know re: `.editorconfig`. Will explore further!

**Comment by pholser** [comment]
:+1: Let me know when ready for merge.

**Comment by ivern** [comment]
Getting there.  Just missing ZonedDateTime now, then need to sort out formatting issues and other cleanup.

**Comment by ivern** [comment]
Ok, I think I'm done.  Let me know what you think.

File: generators/src/main/java/com/pholser/junit/quickcheck/generator/java/time/DayOfWeekGenerator.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import com.pholser.junit.quickcheck.random.SourceOfRandomness;
+
+import java.time.DayOfWeek;
+
+import static com.pholser.junit.quickcheck.internal.Items.choose;
+import static java.util.Arrays.asList;
+
+/**
+ * Produces values of type {@link DayOfWeek}.
+ */
```
Comment (pholser) [inline generators/src/main/java/com/pholser/junit/quickcheck/generator/java/time/DayOfWeekGenerator.java]: Unless used with `From`, custom generators for `enum` types will be trumped by the built-in `EnumGenerator`. I think we can eliminate the `DayOfWeekGenerator` and `MonthGenerator`. It might be good to have tests that demonstrate that you can have these `enum`s as property parameters, though.

**Comment by pholser** [comment]
@ivern Thanks so much for your efforts! This looks really good.

One suggestion, as noted above -- you can probably remove the custom generators for `java.time.*` `enum` types. This is because junit-quickcheck can generate values for `enum` types using the built-in `EnumGenerator` without you having to write a custom generator.

**Comment by ivern** [comment]
Removed, tests added.
